### ISSUE TITLE
Hyperparameters to reproduce the results in paper

### ISSUE THREAD

**Issue by pipipopo** [huawei-noah/DAFL#4]
Hi should we follow the optimizer setting (NAG) in paper or use Adam and SGD in the code? And the learning rate strategy? Using default code i can achieve 84% accuracy on cifar-10 which still can be improved.

**Comment by HantingChen** [comment]
Could you share your log?

**Comment by pipipopo** [comment]
It's something like
[Epoch 1980/2000] [loss_oh: 0.047280] [loss_ie: -0.997264] [loss_a: -0.182967] [loss_kd: 0.088430]
Test Avg. Loss: 0.000497, Accuracy: 0.839200
[Epoch 1981/2000] [loss_oh: 0.042158] [loss_ie: -0.999383] [loss_a: -0.184609] [loss_kd: 0.070008]
Test Avg. Loss: 0.000505, Accuracy: 0.834300
[Epoch 1982/2000] [loss_oh: 0.047642] [loss_ie: -0.997349] [loss_a: -0.183687] [loss_kd: 0.076862]
Test Avg. Loss: 0.000509, Accuracy: 0.834300
[Epoch 1983/2000] [loss_oh: 0.047399] [loss_ie: -0.996971] [loss_a: -0.181911] [loss_kd: 0.090318]
Test Avg. Loss: 0.000542, Accuracy: 0.825300
[Epoch 1984/2000] [loss_oh: 0.046705] [loss_ie: -0.997780] [loss_a: -0.182356] [loss_kd: 0.081486]
Test Avg. Loss: 0.000499, Accuracy: 0.837000
[Epoch 1985/2000] [loss_oh: 0.053331] [loss_ie: -0.996845] [loss_a: -0.183923] [loss_kd: 0.079220]
Test Avg. Loss: 0.000500, Accuracy: 0.836900
[Epoch 1986/2000] [loss_oh: 0.042009] [loss_ie: -0.998083] [loss_a: -0.183647] [loss_kd: 0.063672]
Test Avg. Loss: 0.000494, Accuracy: 0.838000
[Epoch 1987/2000] [loss_oh: 0.041805] [loss_ie: -0.995592] [loss_a: -0.183913] [loss_kd: 0.065907]
Test Avg. Loss: 0.000500, Accuracy: 0.839000
[Epoch 1988/2000] [loss_oh: 0.041465] [loss_ie: -0.996411] [loss_a: -0.183612] [loss_kd: 0.067346]
Test Avg. Loss: 0.000506, Accuracy: 0.835600
[Epoch 1989/2000] [loss_oh: 0.047835] [loss_ie: -0.997881] [loss_a: -0.182845] [loss_kd: 0.081968]
Test Avg. Loss: 0.000519, Accuracy: 0.831000
[Epoch 1990/2000] [loss_oh: 0.044107] [loss_ie: -0.998561] [loss_a: -0.182709] [loss_kd: 0.079596]
Test Avg. Loss: 0.000532, Accuracy: 0.823400
[Epoch 1991/2000] [loss_oh: 0.049824] [loss_ie: -0.997400] [loss_a: -0.181763] [loss_kd: 0.072810]
Test Avg. Loss: 0.000557, Accuracy: 0.818400
[Epoch 1992/2000] [loss_oh: 0.049313] [loss_ie: -0.998589] [loss_a: -0.183052] [loss_kd: 0.080243]
Test Avg. Loss: 0.000560, Accuracy: 0.816700
[Epoch 1993/2000] [loss_oh: 0.045573] [loss_ie: -0.996310] [loss_a: -0.182591] [loss_kd: 0.072807]
Test Avg. Loss: 0.000550, Accuracy: 0.819100
[Epoch 1994/2000] [loss_oh: 0.060042] [loss_ie: -0.998994] [loss_a: -0.180363] [loss_kd: 0.094815]
Test Avg. Loss: 0.000563, Accuracy: 0.813700
[Epoch 1995/2000] [loss_oh: 0.054432] [loss_ie: -0.996170] [loss_a: -0.181537] [loss_kd: 0.077282]
Test Avg. Loss: 0.000550, Accuracy: 0.820500
[Epoch 1996/2000] [loss_oh: 0.043645] [loss_ie: -0.999312] [loss_a: -0.182114] [loss_kd: 0.069092]
Test Avg. Loss: 0.000590, Accuracy: 0.809700
[Epoch 1997/2000] [loss_oh: 0.056552] [loss_ie: -0.997922] [loss_a: -0.181073] [loss_kd: 0.089914]
Test Avg. Loss: 0.000533, Accuracy: 0.826600
[Epoch 1998/2000] [loss_oh: 0.041985] [loss_ie: -0.998161] [loss_a: -0.183008] [loss_kd: 0.078199]
Test Avg. Loss: 0.000542, Accuracy: 0.823100
[Epoch 1999/2000] [loss_oh: 0.050301] [loss_ie: -0.997788] [loss_a: -0.182864] [loss_kd: 0.093478]
Test Avg. Loss: 0.000559, Accuracy: 0.818600

sorry i forget to print this to a file.

**Comment by HantingChen** [comment]
Sorry for the mistake. The 'lr_S' in option should be set as 0.1 on CIFAR-10 and CIFAR-100.

You can run
```shell
python DAFL-train.py --dataset cifar10 --channels 3 --n_epochs 2000 --batch_size 1024 --lr_G 0.02 --lr_S 0.1 --latent_dim 1000  
```
to achieve the results.

**Comment by pipipopo** [comment]
It's even worse. At first accuracy can slowly reach 0.5-0.6, after each adjusting of lr(0.1*), accuracy have a leap of around 0.1, finally at 0.8. Teacher model can get 0.95. I don't have enough gpus so it takes really long to produce results:
[Epoch 1976/2000] [loss_oh: 0.034325] [loss_ie: -0.998383] [loss_a: -0.186675] [loss_kd: 0.044931]
Test Avg. Loss: 0.000730, Accuracy: 0.774600
[Epoch 1977/2000] [loss_oh: 0.042281] [loss_ie: -0.997428] [loss_a: -0.186168] [loss_kd: 0.044020]
Test Avg. Loss: 0.000737, Accuracy: 0.772900
[Epoch 1978/2000] [loss_oh: 0.025967] [loss_ie: -0.998114] [loss_a: -0.186620] [loss_kd: 0.028694]
Test Avg. Loss: 0.000704, Accuracy: 0.784600
[Epoch 1979/2000] [loss_oh: 0.020758] [loss_ie: -0.998541] [loss_a: -0.186852] [loss_kd: 0.029098]
Test Avg. Loss: 0.000746, Accuracy: 0.769600
[Epoch 1980/2000] [loss_oh: 0.019509] [loss_ie: -0.999033] [loss_a: -0.189088] [loss_kd: 0.030632]
Test Avg. Loss: 0.000817, Accuracy: 0.750700
[Epoch 1981/2000] [loss_oh: 0.024321] [loss_ie: -0.998920] [loss_a: -0.191917] [loss_kd: 0.043318]
Test Avg. Loss: 0.000742, Accuracy: 0.774700
[Epoch 1982/2000] [loss_oh: 0.030098] [loss_ie: -0.997930] [loss_a: -0.191318] [loss_kd: 0.045341]
Test Avg. Loss: 0.000758, Accuracy: 0.769100
[Epoch 1983/2000] [loss_oh: 0.029307] [loss_ie: -0.997977] [loss_a: -0.185705] [loss_kd: 0.040635]
Test Avg. Loss: 0.000813, Accuracy: 0.757600
[Epoch 1984/2000] [loss_oh: 0.031141] [loss_ie: -0.997425] [loss_a: -0.189030] [loss_kd: 0.032316]
Test Avg. Loss: 0.000748, Accuracy: 0.771400
[Epoch 1985/2000] [loss_oh: 0.031275] [loss_ie: -0.997536] [loss_a: -0.186609] [loss_kd: 0.046522]
Test Avg. Loss: 0.000704, Accuracy: 0.780500
[Epoch 1986/2000] [loss_oh: 0.027221] [loss_ie: -0.998518] [loss_a: -0.184667] [loss_kd: 0.034934]
Test Avg. Loss: 0.000701, Accuracy: 0.787400
[Epoch 1987/2000] [loss_oh: 0.022373] [loss_ie: -0.997547] [loss_a: -0.185011] [loss_kd: 0.030784]
Test Avg. Loss: 0.000712, Accuracy: 0.780600
[Epoch 1988/2000] [loss_oh: 0.035721] [loss_ie: -0.997044] [loss_a: -0.188190] [loss_kd: 0.049834]
Test Avg. Loss: 0.000695, Accuracy: 0.786200
[Epoch 1989/2000] [loss_oh: 0.031543] [loss_ie: -0.998306] [loss_a: -0.186184] [loss_kd: 0.037074]
Test Avg. Loss: 0.000679, Accuracy: 0.788200
[Epoch 1990/2000] [loss_oh: 0.021992] [loss_ie: -0.997854] [loss_a: -0.185959] [loss_kd: 0.035901]
Test Avg. Loss: 0.000700, Accuracy: 0.782100
[Epoch 1991/2000] [loss_oh: 0.026769] [loss_ie: -0.998564] [loss_a: -0.187954] [loss_kd: 0.042126]
Test Avg. Loss: 0.000703, Accuracy: 0.784700
[Epoch 1992/2000] [loss_oh: 0.026880] [loss_ie: -0.998181] [loss_a: -0.186374] [loss_kd: 0.043251]
Test Avg. Loss: 0.000678, Accuracy: 0.790800
[Epoch 1993/2000] [loss_oh: 0.031500] [loss_ie: -0.998627] [loss_a: -0.185165] [loss_kd: 0.045736]
Test Avg. Loss: 0.000757, Accuracy: 0.767900
[Epoch 1994/2000] [loss_oh: 0.025343] [loss_ie: -0.997458] [loss_a: -0.187835] [loss_kd: 0.036923]
Test Avg. Loss: 0.000797, Accuracy: 0.762600
[Epoch 1995/2000] [loss_oh: 0.023452] [loss_ie: -0.998881] [loss_a: -0.184272] [loss_kd: 0.034962]
Test Avg. Loss: 0.000809, Accuracy: 0.757900
[Epoch 1996/2000] [loss_oh: 0.033766] [loss_ie: -0.998004] [loss_a: -0.185957] [loss_kd: 0.051678]
Test Avg. Loss: 0.000749, Accuracy: 0.772800
[Epoch 1997/2000] [loss_oh: 0.028297] [loss_ie: -0.997233] [loss_a: -0.187246] [loss_kd: 0.042748]
Test Avg. Loss: 0.000814, Accuracy: 0.758200
[Epoch 1998/2000] [loss_oh: 0.033336] [loss_ie: -0.997374] [loss_a: -0.187897] [loss_kd: 0.044380]
Test Avg. Loss: 0.000886, Accuracy: 0.739900
[Epoch 1999/2000] [loss_oh: 0.033546] [loss_ie: -0.998160] [loss_a: -0.188150] [loss_kd: 0.044022]
Test Avg. Loss: 0.000731, Accuracy: 0.777800

**Comment by HantingChen** [comment]
Have you cloned the newest version? I fixed a bug 6 days ago: https://github.com/huawei-noah/DAFL/commit/f5ad6114b5896b918dc5923c99575275ecad41dc.

**Comment by pipipopo** [comment]
Yes, newest version. Can you share your logs?

**Comment by HantingChen** [comment]
[Epoch 1317/2000] [loss_oh: 0.406999] [loss_ie: -0.992317] [loss_a: -0.170399] [loss_kd: 0.154036]
Test Avg. Loss: 0.000335, Accuracy: 0.908900
[Epoch 1318/2000] [loss_oh: 0.409483] [loss_ie: -0.993791] [loss_a: -0.170580] [loss_kd: 0.170466]
Test Avg. Loss: 0.000320, Accuracy: 0.910800
[Epoch 1319/2000] [loss_oh: 0.378921] [loss_ie: -0.996677] [loss_a: -0.168919] [loss_kd: 0.154981]
Test Avg. Loss: 0.000360, Accuracy: 0.903200
[Epoch 1320/2000] [loss_oh: 0.411190] [loss_ie: -0.996795] [loss_a: -0.169832] [loss_kd: 0.153099]
Test Avg. Loss: 0.000326, Accuracy: 0.910500
[Epoch 1321/2000] [loss_oh: 0.388259] [loss_ie: -0.994060] [loss_a: -0.169807] [loss_kd: 0.158724]
Test Avg. Loss: 0.000334, Accuracy: 0.907600
[Epoch 1322/2000] [loss_oh: 0.423804] [loss_ie: -0.993929] [loss_a: -0.169548] [loss_kd: 0.158651]
Test Avg. Loss: 0.000358, Accuracy: 0.903300
[Epoch 1323/2000] [loss_oh: 0.438993] [loss_ie: -0.992474] [loss_a: -0.168553] [loss_kd: 0.163086]
Test Avg. Loss: 0.000326, Accuracy: 0.909300
[Epoch 1324/2000] [loss_oh: 0.418159] [loss_ie: -0.990821] [loss_a: -0.170198] [loss_kd: 0.164858]
Test Avg. Loss: 0.000340, Accuracy: 0.905200
[Epoch 1325/2000] [loss_oh: 0.418637] [loss_ie: -0.993074] [loss_a: -0.169373] [loss_kd: 0.183218]
Test Avg. Loss: 0.000315, Accuracy: 0.914600
[Epoch 1326/2000] [loss_oh: 0.398018] [loss_ie: -0.996654] [loss_a: -0.169380] [loss_kd: 0.156393]
Test Avg. Loss: 0.000320, Accuracy: 0.912300
[Epoch 1327/2000] [loss_oh: 0.423927] [loss_ie: -0.994007] [loss_a: -0.169452] [loss_kd: 0.168334]
Test Avg. Loss: 0.000345, Accuracy: 0.905300
